geomstats/vectorization.py

```diff
@@ -21,78 +20,136 @@ def squeeze_output_dim_0(initial_ndims, point_types):
     - have the corresponding dimension 0 squeezed,
     i.e. if all input parameters have ndim strictly less than the ndim
     corresponding to their vectorized shape.
+
+    Parameters
+    ----------
+    initial_ndims : list
+        Initial ndims of input parameters, as entered by the user.
+    point_types : list
+        Associated list of point_type of input parameters.
+
+    Returns
+    -------
+    squeeze : bool
+        Boolean deciding whether to squeeze dim 0 of the output.
     """
     for ndim, point_type in zip(initial_ndims, point_types):
-        vect_ndim = POINT_TYPES_TO_NDIMS[point_type]
-        assert ndim <= vect_ndim
-        if ndim == vect_ndim:
-            return False
+        if point_type != 'else' and ndim is not None:
+            vect_ndim = POINT_TYPES_TO_NDIMS[point_type]
+            assert ndim <= vect_ndim
+            if ndim == vect_ndim:
+                return False
     return True
 
 
 def is_scalar(vect_array):
+    """Test if an array represents a scalar."""
     has_ndim_2 = vect_array.ndim == 2
+    if not has_ndim_2:
+        return False
     has_singleton_dim_1 = vect_array.shape[1] == 1
-    return has_ndim_2 and has_singleton_dim_1
+    return has_singleton_dim_1
 
 
 def squeeze_output_dim_1(result, initial_shapes, point_types):
     """Determine if the output needs to squeeze a singular dimension 1.
 
     This happens if the user represents scalars as array of shapes:
     [n_samples,] instead of [n_samples, 1]
-
     Dimension 1 is squeezed by default if the return point type is a scalar.
     Dimension 1 is not squeezed if the user inputs at least one scalar with
     a singleton in dimension 1.
+
```

Comment:
The above docstring is not very clear:
- use passive mode for the title: output needs to be squeezed
- why 'return' in return point type?